components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/approles/impl/AppAssociatedRolesResolverImpl.java

```diff
@@ -0,0 +1,348 @@
+
+        return FrameworkServiceDataHolder.getInstance().getRoleManagementServiceV2()
+                .getRoleIdListOfUser(userId, tenantDomain);
+    }
+
+    /**
+     * Get roles associated with the application.
+     *
+     * @param applicationId Application ID.
+     * @param tenantDomain  Tenant domain.
+     * @return Roles associated with the application.
+     * @throws ApplicationRolesException If an error occurred while getting roles associated with the application.
+     */
+    private List<RoleV2> getRolesAssociatedWithApplication(String applicationId, String tenantDomain)
+            throws ApplicationRolesException {
+
+        try {
+            return FrameworkServiceDataHolder.getInstance().getApplicationManagementService()
+                    .getAssociatedRolesOfApplication(applicationId, tenantDomain);
+        } catch (IdentityApplicationManagementException e) {
+            throw RoleResolverUtils.handleServerException(ERROR_CODE_RETRIEVING_APP_ROLES, e);
+        }
+    }
+
+    /**
+     * Get federated user IDP groups.
+     *
+     * @param authenticatedUser Authenticated user.
+     * @return Federated user IDP groups.
+     * @throws ApplicationRolesException If an error occurred while getting federated user IDP groups.
+     */
+    private List<String> getFederatedUserIDPGroup(AuthenticatedUser authenticatedUser)
+            throws ApplicationRolesException {
+
+        String idpName = authenticatedUser.getFederatedIdPName();
+        String tenantDomain = authenticatedUser.getTenantDomain();
+        IdentityProvider federatedIdP;
+        try {
+            federatedIdP = FrameworkServiceDataHolder.getInstance().getIdentityProviderManager()
+                    .getIdPByName(idpName, tenantDomain, true);
+        } catch (IdentityProviderManagementException e) {
+            throw RoleResolverUtils.handleServerException(
+                    ERROR_CODE_RETRIEVING_IDENTITY_PROVIDER, e, idpName, tenantDomain);
+        }
+        if (federatedIdP != null) {
+            String idpGroupsClaimUri = Arrays.stream(federatedIdP.getClaimConfig().getClaimMappings())
+                    .filter(claimMapping ->
+                            FrameworkConstants.GROUPS_CLAIM.equals(claimMapping.getLocalClaim().getClaimUri()))
+                    .map(claimMapping -> claimMapping.getRemoteClaim().getClaimUri())
+                    .findFirst()
+                    .orElse(null);
+
+            if (idpGroupsClaimUri != null) {
+                String[] idpGroups = getIdPUserGroups(authenticatedUser, idpGroupsClaimUri);
+                if (idpGroups != null && idpGroups.length > 0) {
+                    return Arrays.asList(idpGroups);
+                }
+            }
+        }
+        return Collections.emptyList();
+    }
+
+    /**
+     * Get the groups of the local authenticated user.
+     *
+     * @param authenticatedUser Authenticated user.
+     * @return Groups of the local user.
```

Comment:
exception is not in comment

Reply:
Fixed with Fixed with https://github.com/wso2/carbon-identity-framework/pull/5070/commits/cf43977ae1c82e009dda795205668047624a0925